ios/browser/api/certificate/models/brave_certificate_public_key_info.mm

```diff
@@ -0,0 +1,183 @@
+
+    _keyHexEncoded = base::SysUTF8ToNSString(brave::hex_string_from_bytes(
+        static_cast<const std::uint8_t*>([external_representation bytes]),
+        [external_representation length]));
+
+    NSDictionary* attributes =
+        (__bridge_transfer NSDictionary*)SecKeyCopyAttributes(key);
+    if (attributes) {
+      _keySizeInBits =
+          [attributes[(NSString*)kSecAttrKeySizeInBits] integerValue];
+      //_effectiveSize = [(__bridge NSNumber*)attrs[kSecAttrEffectiveKeySize]
+      //integerValue];
+
+      if ([attributes[(NSString*)kSecAttrCanEncrypt] boolValue]) {
+        _keyUsage |= BravePublicKeyUsage_ENCRYPT;
+      }
+
+      if ([attributes[(NSString*)kSecAttrCanDecrypt] boolValue]) {
+        _keyUsage |= BravePublicKeyUsage_DECRYPT;
+      }
+
+      if ([attributes[(NSString*)kSecAttrCanSign] boolValue]) {
+        _keyUsage |= BravePublicKeyUsage_SIGN;
+      }
+
+      if ([attributes[(NSString*)kSecAttrCanVerify] boolValue]) {
+        _keyUsage |= BravePublicKeyUsage_VERIFY;
+      }
+
+      if ([attributes[(NSString*)kSecAttrCanWrap] boolValue]) {
+        _keyUsage |= BravePublicKeyUsage_WRAP;
+      }
+
+      if ([attributes[(NSString*)kSecAttrCanDerive] boolValue]) {
+        _keyUsage |= BravePublicKeyUsage_DERIVE;
+      }
+
+      BravePublicKeyUsage any_usage = 0;
+      any_usage &= BravePublicKeyUsage_ENCRYPT;
+      any_usage &= BravePublicKeyUsage_SIGN;
+      any_usage &= BravePublicKeyUsage_VERIFY;
+      any_usage &= BravePublicKeyUsage_WRAP;
+      any_usage &= BravePublicKeyUsage_DERIVE;
+
+      if (_keyUsage == any_usage) {
+        _keyUsage = BravePublicKeyUsage_ANY;
+      }
+
+      if ([attributes[(NSString*)kSecAttrType]
+              isEqualToString:(NSString*)kSecAttrKeyTypeRSA]) {
+        _type = BravePublicKeyType_RSA;
+        _algorithm = @"RSA";
+
+        // Recreate the SPK without the unused bit count because the previous
+        // remove_prefix code didn't work. Parse the RSA ASN.1 PKCS1 structure
+        // from SecKeyCopyExternalRepresentation.
+        net::der::Input modulus;
+        net::der::Input public_exponent;
+        spk = net::der::Input(
+            static_cast<const std::uint8_t*>([external_representation bytes]),
+            [external_representation length]);
+        if (x509_utils::ParseRSAPublicKeyInfo(spk, &modulus,
+                                              &public_exponent)) {
+          _keyHexEncoded = base::SysUTF8ToNSString(
+              brave::hex_string_from_string(modulus.AsString()));
+
+          std::uint64_t parsed_public_exponent = 0;
+          if (net::der::ParseUint64(public_exponent, &parsed_public_exponent)) {
+            _exponent =
+                static_cast<decltype(_exponent)>(parsed_public_exponent);
+          }
+        }
+      }
+
+      if ([attributes[(NSString*)kSecAttrType]
+              isEqualToString:(NSString*)kSecAttrKeyTypeEC]) {
```

Comment:
No `__bridge` needed?

Reply:
None needed. When I add it, the clang compiler complains. If I use Xcode clang, no complaints. So for now it isn't needed and it compiles fine :)